polus-unet-training-plugin/src/bfio_finetuning.py

```diff
@@ -0,0 +1,455 @@
+
+                        # Loop across the length of the image
+                        for y in range(0,br.Y,tile_size):
+                            y_max = min([br.Y,y+tile_size])
+
+                            # Loop across the depth of the image
+                            for x in range(0,br.X,tile_size):
+                                iofile_path = "_train_"+ str(ind) +".h5"
+                                x_max = min([br.X,x+tile_size])
+                                img = np.squeeze(br[y:y_max,x:x_max,z:z+1,0,0])
+                                roiimage = np.squeeze(br_label[y:y_max,x:x_max,z:z+1,0,0])
+                                if np.unique(roiimage).any()>0:
+                                    _data = normalize(img)
+                                    classlabelsdata, instancelabelsdata, total_instances = createlabelsandweightsfromrois(img, roiimage)
+                                    _weights, _labels, _samplepdf = addLabelsAndWeightsToBlobs(img, classlabelsdata, instancelabelsdata, total_instances, foregroundbackgroundratio, borderWeightFactor, borderWeightSigmaPx, sigma1Px)
+                                    saveBlobs(_data, className, iofile_path, _weights, _labels, _samplepdf, foregroundbackgroundratio, borderWeightFactor, borderWeightSigmaPx, sigma1Px)
+                                    ind+=1
+
+    finally:
+        logger.info("training blobs created.")
+
+
+    filename = 'trainfilelist.txt'
+    trainingfile = open(filename, "w+")
+    for i in range(ind):
+        trainingfile.write("_train_"+str(i)+".h5"+"\n")
+    trainingfile.close()
+
+    ## Validation Images
+    rootdir2 = Path(testingImages)
+    filename = 'validfilelist.txt'
+    validationfile = open(filename, "w+")
+    total_valid = 0; ind = 0
+    try:     
+        filepath = pathlib.Path(__file__).parent
+        filepath = filepath.joinpath(rootdir2)
+        # i = 0
+        for file in filepath.iterdir():
+            img_path = Path(file)
+            mask_path = Path(testingLabels+"/"+file.name)
+            logger.info('Processing validation image {}'.format(img_path))
+            logger.info('Processing validation mask {}'.format(mask_path))
+            tile_grid_size = 1
+            tile_size = tile_grid_size * 1024
+            with BioReader(img_path,backend='python',max_workers=cpu_count()) as br:
+                with BioReader(mask_path,backend='python',max_workers=cpu_count()) as br_label:
+                    # Loop through z-slices
+                    for z in range(br.Z):
+
+                        # Loop across the length of the image
+                        for y in range(0,br.Y,tile_size):
+                            y_max = min([br.Y,y+tile_size])
+
+                            # Loop across the depth of the image
+                            for x in range(0,br.X,tile_size):
+                                valid_name = "_valid_"+str(ind) + "_"
+                                x_max = min([br.X,x+tile_size])
```

Comment:
This seems like it's identical code to what we see in the training images section of code. Can we remove the redundancy somehow? This will make the code easier to read and maintain in the long run.